samcli/lib/observability/cw_logs/cw_log_puller.py

```diff
@@ -0,0 +1,111 @@
+"""
+CloudWatch log event puller implementation
+"""
+import logging
+import time
+from datetime import datetime
+from typing import Optional, Any
+
+from samcli.lib.observability.cw_logs.cw_log_event import CWLogEvent
+from samcli.lib.observability.observability_info_puller import ObservabilityPuller, ObservabilityEventConsumer
+from samcli.lib.utils.time import to_timestamp, to_datetime
+
+LOG = logging.getLogger(__name__)
+
+
+class CWLogPuller(ObservabilityPuller):
+    """
+    Puller implementation that can pull events from CloudWatch log group
+    """
+
+    def __init__(
+        self,
+        logs_client: Any,
+        consumer: ObservabilityEventConsumer,
+        cw_log_group: str,
+        resource_name: Optional[str] = None,
+        max_retries: int = 1000,
```

Comment:
This is a large number of retries, is there any reason for it?

Reply:
Not sure, I just refactored the current value from here; https://github.com/aws/aws-sam-cli/blob/develop/samcli/lib/logs/fetcher.py#L81

AFAIK, this tracks the latest log messages, so if there are no new log events for 1000 retry, the command will exit. I agree this is bad for both cases, whether you get an error or when you don't get any new logging message. I will check with UX to see if we need to tie this to a parameter to make it more flexible.